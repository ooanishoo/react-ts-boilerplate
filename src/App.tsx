import React from 'react'
import { Header, Main } from './components'

const App: React.FC = () => {
    return (
        <>
            <Header />
            <Main />
        </>
    )
}

export default App
