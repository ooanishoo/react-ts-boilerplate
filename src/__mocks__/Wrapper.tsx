import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'

/* a wrapper Router component that takes any React component and renders as it's child component */
type WrapperProps = {
    component?: React.ReactNode
}
export const Wrapper: React.FC<WrapperProps> = ({ component }: WrapperProps) => <Router>{component}</Router>
