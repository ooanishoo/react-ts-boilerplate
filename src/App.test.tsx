import { cleanup, render } from '@testing-library/react'
import React from 'react'
import App from './App'
import { Wrapper } from './__mocks__/Wrapper'

afterEach(cleanup)

describe('<App />', () => {
    test('renders without crashing', () => {
        const { container } = render(<Wrapper component={<App />} />)
        expect(container).toMatchSnapshot()
    })
})
