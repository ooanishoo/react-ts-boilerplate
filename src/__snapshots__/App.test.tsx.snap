// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders without crashing 1`] = `
<div>
  <header
    data-testid="header"
  >
    <h1>
      My Project
    </h1>
    <nav
      data-testid="navigation"
      role="navigation"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/todos"
          >
            Todos
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    data-testid="main"
    role="main"
  >
    <div
      data-testid="home"
    >
      This is a Home Page
    </div>
  </main>
</div>
`;
